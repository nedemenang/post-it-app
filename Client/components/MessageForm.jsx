import React, {Component} from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import '../public/style.css';
import $ from '../public/jquery.js';
import AppActions from '../actions/AppActions';
import ls from 'local-storage';

class MessageForm extends Component {

submit(event){
  event.preventDefault();

  let today = new Date();
  let dd = today.getDate();
  let mm = today.getMonth()+1; //January is 0!
  let yyyy = today.getFullYear();

  if(dd<10) {
      dd = '0'+dd
  } 

  if(mm<10) {
      mm = '0'+mm
  } 

  today = mm + '/' + dd + '/' + yyyy;

  const messagebody = this.refs.message.value.trim();
  const priority = this.refs.priority.value.trim();
  const postedon = today;
  
  //console.log(this.props.selectedGroupId);
  
  if(this.props.selectedGroup.length === 0)
  {
    AppActions.receiveErrors('Please select a group to post a message');
    //console.log(this.props.errors);

  }else if(this.refs.message.value === '')
  {
    AppActions.receiveErrors('Please type in a message');
  }
  else if (this.refs.priority.value === 'Select Message Priority ....')
  {
    AppActions.receiveErrors('Please select a message priority');
  }else{
    //console.log(ls.get('user'));
    const user = ls.get('user');
    //console.log(user[0].email);
    let messageObject = {
      messageBody : messagebody,
      postedon: postedon,
      priority: priority,
      postedBy: user[0].email, //this.props.loggedInUser[0].email,
      postedByDisplayName: user[0].displayName,//this.props.loggedInUser[0].displayName,
      profilePic: user[0].profilePic,
      groupId: this.props.selectedGroup[0].groupId,
    }
    console.log(messageObject);

    AppActions.addMessage(messageObject);
    this.refs.message.value = '';
    this.refs.priority.value = 'Select Message Priority ....';
    AppActions.receiveErrors('');
  }
  
}

  constructor(props){
    super(props);
    this.state= {};
    this.submit = this.submit.bind(this);
  }
  render(){
    return(
      <div>
         <form>
           <input type="text" className="form-control" ref="message" placeholder="Please type a message. Press enter to submit." />
            <select ref="priority" className="form-select">
              <option>Select Message Priority ....</option>
              <option value="normal">Normal</option>
              <option value="urgent">Urgent</option>
              <option value="critical">Critical</option>
             </select>
             <p className="error">{this.props.errors}</p>
              <button className="messageButton" onClick={this.submit}>Submit</button>
         </form>
      </div>
    );
  }
}

export default MessageForm;