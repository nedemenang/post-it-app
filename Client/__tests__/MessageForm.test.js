import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MessageForm from '../components/MessageForm.jsx';

describe('the Message form component', () => {
  let mountedComponent;
  jest.mock('../dispatcher/AppDispatcher');

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: `${str2}static/files/blank-profile-pic.png`
  };

  const group = {
    groupId: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    groupname: 'Test Group'
  }

  const messageFormItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<MessageForm loggedInUser = {loggedInUser} selectedGroup={group} key = {1}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = messageFormItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = messageFormItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should recieve props', () => {
    expect(Object.keys(messageFormItem().props()).length).toBeGreaterThan(0);
  });

  it('should call function on submit of form', () => {
    const component = shallow(<MessageForm loggedInUser = {loggedInUser}/>);
    const preventDefault = jest.fn();
    component.find('button').simulate('click', { preventDefault });
    expect(preventDefault).toBeCalled();
  });

  it('should update state on message field change', () => {
    messageFormItem().find('#message').simulate('change', { target: {
      value: 'New message'
    } });
    expect(messageFormItem().state().message).toBe('New message');
  });

  it('should update state on priority field change', () => {
    messageFormItem().find('.form-select').simulate('change', { target: {
      value: 'urgent'
    } });
    expect(messageFormItem().state().priority).toBe('urgent');
  });
});
