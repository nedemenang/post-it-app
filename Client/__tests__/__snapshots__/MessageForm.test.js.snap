// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Message form component should match snapshot test 1`] = `
<div
  className="footer"
>
  <form
    id="chatform"
  >
    <select
      className="form-select"
      onChange={[Function]}
      value="normal"
    >
      <option
        value="normal"
      >
        Normal
      </option>
      <option
        value="urgent"
      >
        Urgent
      </option>
      <option
        value="critical"
      >
        Critical
      </option>
    </select>
    <textarea
      id="message"
      onChange={[Function]}
      placeholder="Please type a message."
      type="text"
      value=""
    />
    <FlatButton
      disabled={false}
      fullWidth={false}
      icon={
        <Icon
          fill="currentColor"
          icon={
            Object {
              "children": Array [
                Object {
                  "attribs": Object {
                    "d": "M1764 11q33 24 27 64l-256 1536q-5 29-32 45-14 8-31 8-11 0-24-5l-453-185-242 295q-18 23-49 23-13 0-22-4-19-7-30.5-23.5t-11.5-36.5v-349l864-1059-1069 925-395-162q-37-14-40-55-2-40 32-59l1664-960q15-9 32-9 20 0 36 11z",
                  },
                  "name": "path",
                },
              ],
              "viewBox": "0 0 1792 1792",
            }
          }
          size={16}
          style={
            Object {
              "color": "#4CAF4F",
            }
          }
        />
      }
      labelPosition="after"
      labelStyle={Object {}}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "minWidth": 35,
        }
      }
    />
  </form>
</div>
`;
